### PR TITLE
Update special attack plugin to include ToB damage modifiers

### LINKED ISSUE #9862: BGS Damage counter is incorrect in the theatre of blood
XP modifiers are static so this should be fixable

### PR DESCRIPTION
**Pull request by winterdaze**
Closes #9862

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/Varbits.java [hunk 1]
```diff
@@ -302,6 +302,15 @@
 	 */
 	THEATRE_OF_BLOOD(6440),
 
+	/**
+	 * Theatre of Blood orb varbits each number stands for the player's health on a scale of 1-27 (I think), 0 hides the orb
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/Varbits.java]: `I think`?

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java [hunk 2]
```diff
@@ -56,6 +59,7 @@
 import net.runelite.client.ws.WSClient;
 import org.apache.commons.lang3.ArrayUtils;
 
+
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java]: Remove this extra newline
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java [hunk 3]
```diff
@@ -232,12 +242,34 @@ private void addInteracting(int npcId)
...
 		if (boss != null)
 		{
 			modifier = boss.getModifier();
-			interactedNpcIds.addAll(boss.getIds());
+			if (boss.isTob())
+			{
+				int partySize = getTobPartySize();
+				modifier = boss.getModifier(partySize);
+			}
+			interactedNpcIds.addAll(Sets.newHashSet(Arrays.stream(boss.getIds()).boxed().toArray(Integer[]::new)));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java]:
```suggestion
			interactedNpcIds.addAll(Arrays.stream(boss.getIds()).boxed().collect(Collectors.toSet()));
```

Or just add the IDs in a loop:

```java
for (int bossId : boss.getIds())
{
	interactedNpcIds.add(bossId);
}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java [hunk 4]
```diff
@@ -232,12 +242,34 @@ private void addInteracting(int npcId)
...
-		final Boss boss = Boss.getBoss(npcId);
+		final Boss boss = Boss.findBoss(npcId);
 		if (boss != null)
 		{
 			modifier = boss.getModifier();
-			interactedNpcIds.addAll(boss.getIds());
+			if (boss.isTob())
+			{
+				int partySize = getTobPartySize();
+				modifier = boss.getModifier(partySize);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java]: `partySize` isn't reused; pass it directly to `boss.getModifier()`.
```suggestion
				modifier = boss.getModifier(getTobPartySize());
```

File: runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java [hunk 5]
```diff
@@ -34,39 +34,89 @@
...
+	Boss(float[] modifiers, int... ids)
 	{
-		this.modifier = modifier;
-		this.ids = Sets.newHashSet(ids);
+		this.modifier = modifiers;
+		this.ids = ids;
+		this.isTob = true;
 	}
 
-	static Boss getBoss(int id)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java]: Why change the method name?

**Comment by Adam-** [comment]
this is unnecessary now due to 8674a4f998ada7708de842ec98e9cc098b4c385c

**Pull request closed without merging**